src/cov.jl

```diff
@@ -148,8 +163,10 @@ standard deviations `s`.
 function cor2cov!(C::AbstractMatrix, s::AbstractArray)
     n = length(s)
     size(C) == (n, n) || throw(DimensionMismatch("inconsistent dimensions"))
-    for i in CartesianIndices(size(C))
-        @inbounds C[i] *= s[i[1]] * s[i[2]]
+    @inbounds for i in CartesianIndices(size(C))
+        si = s[i[1]] * s[i[2]]
+        # the covariance is 0 when si==0, although C[i] is NaN in this case
+        C[i] = iszero(si) ? zero(eltype(C)) : C[i] * si
```

Comment:
In case `eltype(C)` isn't a concrete type:
```suggestion
        Ci = C[i]
        C[i] = iszero(si) ? zero(Ci) : Ci * si
```
Can you explain a bit more what happens here?

Reply:
If one of the variables has zero variance, its covariance with any other variable is 0, but the correlation is undefined, and the result of `cor` would have `NaN` in that element. In that case `Ci == NaN`, `si == 0`, but the correct output is 0, not `NaN`.
For example, 
```
X = [1:3 ones(3)]
cov(X) == [1 0; 0 0]
cor(X) == [1 NaN; NaN 1]
cor2cov(cor(X), std.(eachcol(X))) # previously [1 NaN; NaN 0], which is incorrect
```